mesonbuild/mesonlib.py

```diff
@@ -308,7 +308,22 @@ def __lt__(self, other):
             return self.value < other.value
         return NotImplemented
 
-MachineChoice = OrderedEnum('MachineChoice', ['BUILD', 'HOST', 'TARGET'])
+class MachineChoice(OrderedEnum):
+
+    """Enum class representing one of the three possible values for binaries,
+    the build, host, and target machines.
+    """
+
+    BUILD = 0
+    HOST = 1
+    TARGET = 2
+
+    @classmethod
+    def is_native(cls, native: bool) -> 'MachineChoice':
```

Comment:
Please do not add this. You don't need this in the one place you used it above, and in https://github.com/mesonbuild/meson/pull/4010 which is almost next in my serious, I remove most of the other usecases where we might use this.

`native` keyword args in Meson are not yet used consistently, so we cannot abstract over the conversion from those either.

Reply:
Yes, they should just about all be, but now that you made the change I'm happy to just do that as part of #4010.

Reply:
Okay, then I'll drop these patches.

Reply:
I mean it's fine either way, #4010 will probably land slowly.